### PR TITLE
Add org.jflap.JFLAP

### PR DESCRIPTION
**Pull request by reisaraujo-miguel**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.reisaraujo_miguel.JFLAP.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2023 Miguel R. Araújo -->
+<component type="desktop-application">
+  <id>io.github.reisaraujo_miguel.JFLAP</id>
```
Comment (hfiguiere) [inline io.github.reisaraujo_miguel.JFLAP.metainfo.xml:4]: why not `org.jflap.jflap`?

File: io.github.reisaraujo_miguel.JFLAP.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://www2.cs.duke.edu/csed/jflap/jflaptmp/july27-18/license.html</project_license>
+  <url type="homepage">https://www.jflap.org/</url>
+  <url type="help">https://www.jflap.org/tutorial/</url>
+  <url type="contact">https://www.jflap.org/contactinfo.html</url>
+  
+  <description>
+    <p>
+      JFLAP is software for experimenting with formal languages topics including nondeterministic finite automata, nondeterministic pushdown automata, multi-tape Turing machines, several types of grammars, parsing, and L-systems. In addition to constructing and testing examples for these, JFLAP allows one to experiment with construction proofs from one form to another, such as converting an NFA to a DFA to a minimal state DFA to a regular expression or regular grammar.
+    </p>
```
Comment (hfiguiere) [inline io.github.reisaraujo_miguel.JFLAP.metainfo.xml:18]: this also need a disclaimer that it is unofficial

File: io.github.reisaraujo_miguel.JFLAP.metainfo.xml (hunk 1 above)
Comment (reisaraujo-miguel) [reply io.github.reisaraujo_miguel.JFLAP.metainfo.xml:4]: I do not own this domain

Comment (hfiguiere) [reply io.github.reisaraujo_miguel.JFLAP.metainfo.xml:4]: upstream does.

Comment (reisaraujo-miguel) [reply io.github.reisaraujo_miguel.JFLAP.metainfo.xml]: Fixed

File: images/finite-automaton.png
Comment (bbhtt) [inline images/finite-automaton.png:1]: I don't see where these 4 screenshots are being used?

**Comment by bbhtt** [comment]
bot, build org.jflap.JFLAP

File: org.jflap.JFLAP.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2023 Miguel R. Araújo -->
+<component type="desktop-application">
+  <id>org.jflap.JFLAP</id>
+  
+  <name>JFLAP</name>
+  <summary>Formal languages testing</summary>
+  
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://www2.cs.duke.edu/csed/jflap/jflaptmp/july27-18/license.html</project_license>
```
Comment (bbhtt) [inline org.jflap.JFLAP.metainfo.xml:10]: Add developer_name

**Comment by bbhtt** [comment]
Not sure why it's failing on CI, linter is passing locally.

**Comment by reisaraujo-miguel** [comment]
> Not sure why it's failing on CI, linter is passing locally.

There is some problem with the screenshots, it's saying there is no screenshot when clearly there is.

I'm on fedora, when I use the rpm package to build and install there is no problem with the screenshots, but when I use the flatpak  version of flatpak-builder with these commands:
```sh
flatpak run org.flatpak.Builder --force-clean --sandbox --user --install --install-deps-from=flathub --ccache --mirror-screenshots-url=https://dl.flathub.org/repo/screenshots --repo=repo builddir org.jflap.JFLAP.yml
```
I get the following errors on the linter:
```json
{
    "errors": [
        "appstream-missing-screenshots",
        "appstream-failed-validation"
    ],
    "warnings": [
        "appstream-missing-developer-name"
    ]
}
```
And when I go to the app details on gnome-software the images are not shown:
image

PS: I removed `xml:lang="en"` from <screenshot> because when I add this I don't even get these icons where the images should be, it just says there are no screenshots.

**Comment by reisaraujo-miguel** [comment]
bot, build org.jflap.JFLAP

**Comment by bbhtt** [comment]
Hmm, good catch, xml:lang is supported for the image tag, not screenshots

**Comment by reisaraujo-miguel** [comment]
> Hmm, good catch, xml:lang is supported for the image tag, not screenshots

This should be fixed in the docs, it is saying to use this tag on \<screenshot\>
image

**Comment by bbhtt** [comment]
Thanks I'll fix it.

**Comment by barthalion** [comment]
bot, build org.jflap.JFLAP

**Pull request closed without merging**